pyseir/load_data.py

```diff
@@ -508,44 +503,185 @@ def load_hospitalization_data_by_state(
+    Parameters
+    ----------
+    state: str
+        US state to lookup.
+    t0: datetime
+        Datetime to offset by.
+    category: str
+        'icu' for just ICU or 'hospitalized' for all ICU + Acute.
+
+    Returns
+    -------
+    time: float
+        Days since t0 for the hospitalization data.
+    current estimate: float
+        The most recent estimate for the current occupied in the requested category.
+    """
+    MIN_DATAPOINTS_TO_CONVERT = 3  # We wait until the third datapoint to have 2 deltas to forecast
+
+    abbr = us.states.lookup(state).abbr
+    df = (
+        combined_datasets.build_us_timeseries_with_all_fields()
+        .get_subset(AggregationLevel.STATE, country="USA", state=abbr)
+        .get_data(country="USA", state=abbr)
+    )
+
+    categories = ["icu", "hospitalized"]
+    if category not in categories:
+        raise ValueError(f"Hospitalization category {category} is not in {categories}")
 
-        if convert_cumulative_to_current:
-            # Must be here to avoid circular import. This is required to convert
-            # cumulative hosps to current hosps. We also just use a dummy fips and t_list.
-            from pyseir.parameters.parameter_ensemble_generator import ParameterEnsembleGenerator
-
-            params = ParameterEnsembleGenerator(
-                fips="06", t_list=[], N_samples=1
-            ).get_average_seir_parameters()
-            if category == "hospitalized":
-                average_length_of_stay = (
-                    params["hospitalization_rate_general"]
-                    * params["hospitalization_length_of_stay_general"]
-                    + params["hospitalization_rate_icu"]
-                    * (1 - params["fraction_icu_requiring_ventilator"])
-                    * params["hospitalization_length_of_stay_icu"]
-                    + params["hospitalization_rate_icu"]
-                    * params["fraction_icu_requiring_ventilator"]
-                    * params["hospitalization_length_of_stay_icu_and_ventilator"]
-                ) / (params["hospitalization_rate_general"] + params["hospitalization_rate_icu"])
-            else:
-                average_length_of_stay = (
-                    (1 - params["fraction_icu_requiring_ventilator"])
-                    * params["hospitalization_length_of_stay_icu"]
-                    + params["fraction_icu_requiring_ventilator"]
-                    * params["hospitalization_length_of_stay_icu_and_ventilator"]
-                )
-
-            # Now compute a cumulative sum, but at each day, subtract the discharges from the previous count.
-            new_hospitalizations = np.append([0], np.diff(cumulative))
-            current = [0]
-            for i, new_hosps in enumerate(new_hospitalizations[1:]):
-                current.append(current[i] + new_hosps - current[i] / average_length_of_stay)
-            return times_new, current, HospitalizationDataType.CURRENT_HOSPITALIZATIONS
+    if len(df) == 0:
```

Comment:
not sure if you run into this, but I've seen a lot of 

`if df is None or if df.empty` in the code. This check will raise an exception if it gets a `None`